docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -425,6 +425,29 @@ If a canister traps or panics in `pre_upgrade`, this can lead to permanently blo
 
 - See [Current limitations of the Internet Computer](https://wiki.internetcomputer.org/wiki/Current_limitations_of_the_Internet_Computer), section "Bugs in `pre_upgrade` hooks"
 
+### Reinstantiate global timers during upgrades
+
+#### Security Concern
+
+Global timers are deactivated upon changes to the canister's Wasm module. The [IC specification](https://internetcomputer.org/docs/current/references/ic-interface-spec#timer) states this as follows:
+
+> The global timer is also deactivated upon changes to the canister's Wasm module (calling install_code, uninstall_code methods of the management canister or if the canister runs out of cycles). In particular, the function canister_global_timer won't be scheduled again unless the canister sets the global timer again (using the System API function ic0.global_timer_set).
+
+Upgrade is a mode of `install_code` and hence the global timers are deactivated during an upgrade.
```

Comment:
Since this describes a "Security Concern", would it make sense to elaborate more what bad consequences for security this could have? E.g. along these lines: 
```
If such a timer is used to manage regular operations such as cleanups or payments, this can be risky from a security perspective. 
```
Maybe a good example would be what could have gone wrong in ckBTC (I don't remember)?

Reply:
For example, a timer used to update the exchange rate of a trading pair on a DeX.

Reply:
```suggestion
Upgrade is a mode of `install_code` and hence the timers are deactivated during an upgrade.

This could result in a vulnerability in certain cases where security controls or other critical features rely on these timers. For example, a DEX which relies on timers to update the exchange rates of currencies could be vulnerable to arbitraging opportunities if the rates are no longer updated.
```